drucker_dashboard/apis/api_evaluation.py

```diff
@@ -0,0 +1,191 @@
+
+    @eval_info_namespace.expect(upload_parser)
+    @eval_info_namespace.marshal_with(eval_data_upload)
+    def post(self, application_id:int):
+        """update data to be evaluated"""
+        args = self.upload_parser.parse_args()
+        file = args['file']
+        checksum = HashUtil.checksum(file)
+
+        eobj = db.session.query(Evaluation).filter(
+            Evaluation.application_id == application_id,
+            Evaluation.checksum == checksum).one_or_none()
+        if eobj is not None:
+            return {"status": True, "evaluation_id": eobj.evaluation_id}
+
+        eval_data_path = "eval-{0:%Y%m%d%H%M%S}.txt".format(datetime.datetime.utcnow())
+
+        sobj = Service.query.filter_by(application_id=application_id).first_or_404()
+
+        drucker_dashboard_application = DruckerDashboardClient(logger=api.logger, host=sobj.host)
+        response_body = drucker_dashboard_application.run_upload_evaluation_data(file, eval_data_path)
+
+        if not response_body['status']:
+            raise Exception('Failed to upload')
+        eobj = Evaluation(checksum=checksum, application_id=application_id, data_path=eval_data_path)
+        db.session.add(eobj)
+        db.session.flush()
+        evaluation_id = eobj.evaluation_id
+        db.session.commit()
+        db.session.close()
+
+        return {"status": True, "evaluation_id": evaluation_id}
+
+
+@eval_info_namespace.route('/<int:application_id>/evaluation/<int:evaluation_id>')
+class ApiEvaluation(Resource):
+
+    @eval_info_namespace.marshal_with(success_or_not)
+    def delete(self, application_id:int, evaluation_id:int):
+        """delete data to be evaluated"""
+        eval_query = db.session.query(Evaluation)\
+            .filter(Evaluation.application_id == application_id,
+                    Evaluation.evaluation_id == evaluation_id)
+        if eval_query.one_or_none() is None:
+            return {"status": False, "message": "Not Found."}, 404
+
+        eval_query.delete()
+        db.session.query(EvaluationResult)\
+            .filter(EvaluationResult.evaluation_id == evaluation_id).delete()
+        db.session.commit()
+        db.session.close()
+
+        return {"status": True, "message": "Success."}
+
+
+@eval_info_namespace.route('/<int:application_id>/evaluate')
+class ApiEvaluate(Resource):
+    eval_parser = reqparse.RequestParser()
+    eval_parser.add_argument('service_id', location='form', type=int, required=True)
+    eval_parser.add_argument('evaluation_id', location='form', type=int, required=False)
+    eval_parser.add_argument('overwrite', location='form', type=bool, required=False)
+
+    @eval_info_namespace.expect(eval_parser)
+    @eval_info_namespace.marshal_with(eval_metrics)
+    def post(self, application_id:int):
+        """evaluate"""
+        args = self.eval_parser.parse_args()
+        eval_id = args.get('evaluation_id', None)
+        service_id = args.get('service_id')
```

Comment:
The combination of `evaluation_id` and `model_id` will be better for UX.
To realize it, we need some features but at the moment how about the following spec?

1. Parameters are `evaluation_id` and `model_id` (I prefer to use path parameter for `id` but it doesn't matter)
1. Raise errors when specifying `model_id` which are not used on services.
1. Raise errors when specifying production services.

Reply:
Great work!